enterprise/api_client/sso_orchestrator.py

```diff
@@ -0,0 +1,140 @@
+import base64
+import requests
+
+from http import HTTPStatus
+from urllib.parse import urljoin
+
+from rest_framework.reverse import reverse
+
+from django.conf import settings
+
+from enterprise.errors import ClientError
+from enterprise.utils import (
+    get_configuration_value,
+    get_sso_orchestrator_api_base_url,
+    get_sso_orchestrator_configure_path,
+    get_sso_orchestrator_basic_auth_username,
+    get_sso_orchestrator_basic_auth_password,
+)
+
+class EnterpriseSSOOrchestratorApiClient:
+    """
+    The enterprise API client to communicate with the SSO Orchestrator API. Reads conf settings values to determine
+    orchestration paths and credentials. 
+    
+    Required settings values:
+    - LMS_ROOT_URL
+    - SSO_ORCHESTRATOR_API_BASE_URL
+    - SSO_ORCHESTRATOR_CONFIGURE_PATH
+    - SSO_ORCHESTRATOR_BASIC_AUTH_USERNAME
+    - SSO_ORCHESTRATOR_BASIC_AUTH_PASSWORD
+    """
+
+    def __init__(self):
+        self.base_url = get_sso_orchestrator_api_base_url()
+        self.session = None
+
+    def _get_orchestrator_callback_url(self, config_pk):
+        """
+        get the callback url for the SSO Orchestrator API
+        """
+        lms_base_url = get_configuration_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)
```

Comment:
is this the correct way to call `get_configuration_value`? it doesn't seem correct to me to supply the setting as the default value for when the setting does not exist? perhaps i misunderstand

Reply:
it's how we've done it in other areas

Reply:
https://github.com/openedx/edx-enterprise/blob/426413c29c07a14a2b5a971255c63d1e58f4796d/enterprise/models.py#L625-L661